SilKit/source/experimental/netsim/SimulatedNetworkInternal.cpp

```diff
@@ -57,21 +58,33 @@ void SimulatedNetworkInternal::CreateAndSetEventProducer()
     }
 }
 
+auto SimulatedNetworkInternal::ExtractControllerTypeName(const SilKit::Core::ServiceDescriptor& serviceDescriptor)
+    -> std::string
+{
+    std::string controllerTypeName = "";
+    if (!serviceDescriptor.GetSupplementalDataItem(SilKit::Core::Discovery::controllerType, controllerTypeName))
+    {
+        throw LogicError("No controller type defined in supplemental data.");
+    }
+    return controllerTypeName;
+}
+
 void SimulatedNetworkInternal::AddSimulatedController(const SilKit::Core::ServiceDescriptor& serviceDescriptor,
                                                       ControllerDescriptor controllerDescriptor)
 {
     auto controllerName = serviceDescriptor.GetServiceName();
     auto fromParticipantName = serviceDescriptor.GetParticipantName();
     auto serviceId = serviceDescriptor.GetServiceId();
-    auto serviceDescriptorStr = serviceDescriptor.to_string();
 
     _controllerDescriptors[fromParticipantName][serviceId] = controllerDescriptor;
 
     auto userSimulatedController = _userSimulatedNetwork->ProvideSimulatedController(controllerDescriptor);
 
     if (userSimulatedController)
     {
-        _simulatedNetworkRouter->AddSimulatedController(fromParticipantName, controllerName, serviceId,
+        std::string controllerTypeName = ExtractControllerTypeName(serviceDescriptor);
```

Comment:
couldn't we make this call inside of AddSimulatedController? we're passing serviceDescriptor there anyways. Or am I overlooking some other usages of this?